cmd/create/service/cmd.go

```diff
@@ -0,0 +1,360 @@
+/*
+Copyright (c) 2022 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package service
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/spf13/cobra"
+
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/openshift/rosa/pkg/arguments"
+	"github.com/openshift/rosa/pkg/aws"
+	"github.com/openshift/rosa/pkg/info"
+	"github.com/openshift/rosa/pkg/logging"
+	"github.com/openshift/rosa/pkg/ocm"
+	"github.com/openshift/rosa/pkg/output"
+	"github.com/openshift/rosa/pkg/properties"
+	rprtr "github.com/openshift/rosa/pkg/reporter"
+)
+
+var args ocm.CreateManagedServiceArgs
+
+var Cmd = &cobra.Command{
+	Use:     "service",
+	Aliases: []string{"appliance"},
+	Short:   "Creates a managed service.",
+	Long: `  Managed Services are OpenShift clusters that provide a specific function.
+  Use this command to create managed services.`,
+	Example: `  # Create a Managed Service of type service1.
+  rosa create service --type=service1 --name=clusterName`,
+	Run:                run,
+	Hidden:             true,
+	DisableFlagParsing: true,
+	Args: func(cmd *cobra.Command, argv []string) error {
+		err := arguments.ParseUnknownFlags(cmd, argv)
+		if err != nil {
+			return err
+		}
+
+		if len(cmd.Flags().Args()) > 0 {
+			return fmt.Errorf("Unrecognized command line parameter")
+		}
+		return nil
+	},
+}
+
+func init() {
+	flags := Cmd.Flags()
+	flags.SortFlags = false
+
+	// Basic options
+	flags.StringVar(
+		&args.ServiceType,
+		"type",
+		"",
+		"Type of service.",
+	)
+
+	flags.StringVar(
+		&args.ClusterName,
+		"name",
+		"",
+		"Name of the service instance.",
+	)
```

Comment:
Should we make these as required, so that we can validate in the CLI the request is as expected before sending it to the API. At the moment its a little weird the flow and not overly clear in the error IMO 

```
$ rosa create service 
W: More than one Installer role found, using "arn:aws:iam::765374464689:role/ManagedOpenShift-Installer-Role"
I: Using "arn:aws:iam::765374464689:role/ManagedOpenShift-Worker-Role" for the Worker role
I: Using "arn:aws:iam::765374464689:role/ManagedOpenShift-Support-Role" for the Support role
I: Using "arn:aws:iam::765374464689:role/ManagedOpenShift-ControlPlane-Role" for the ControlPlane role
E: Failed to create managed service: service is required

```